In [1]:
import numpy as np
import pandas as pd

In [2]:
data = pd.read_csv('archive/WA_Fn-UseC_-Telco-Customer-Churn.csv')

In [3]:
data.head()

customerID  gender  SeniorCitizen Partner Dependents  tenure PhoneService  \
0  7590-VHVEG  Female              0     Yes         No       1           No   
1  5575-GNVDE    Male              0      No         No      34          Yes   
2  3668-QPYBK    Male              0      No         No       2          Yes   
3  7795-CFOCW    Male              0      No         No      45           No   
4  9237-HQITU  Female              0      No         No       2          Yes   

      MultipleLines InternetService OnlineSecurity  ... DeviceProtection  \
0  No phone service             DSL             No  ...               No   
1                No             DSL            Yes  ...              Yes   
2                No             DSL            Yes  ...               No   
3  No phone service             DSL            Yes  ...              Yes   
4                No     Fiber optic             No  ...               No   

  TechSupport StreamingTV StreamingMovies        Contract PaperlessBilling  \
0          No          No              No  Month-to-month              Yes   
1          No          No              No        One year               No   
2          No          No              No  Month-to-month              Yes   
3         Yes          No              No        One year               No   
4          No          No              No  Month-to-month              Yes   

               PaymentMethod MonthlyCharges  TotalCharges Churn  
0           Electronic check          29.85         29.85    No  
1               Mailed check          56.95        1889.5    No  
2               Mailed check          53.85        108.15   Yes  
3  Bank transfer (automatic)          42.30       1840.75    No  
4           Electronic check          70.70        151.65   Yes  

[5 rows x 21 columns]

In [5]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7043 entries, 0 to 7042
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   customerID        7043 non-null   object 
 1   gender            7043 non-null   object 
 2   SeniorCitizen     7043 non-null   int64  
 3   Partner           7043 non-null   object 
 4   Dependents        7043 non-null   object 
 5   tenure            7043 non-null   int64  
 6   PhoneService      7043 non-null   object 
 7   MultipleLines     7043 non-null   object 
 8   InternetService   7043 non-null   object 
 9   OnlineSecurity    7043 non-null   object 
 10  OnlineBackup      7043 non-null   object 
 11  DeviceProtection  7043 non-null   object 
 12  TechSupport       7043 non-null   object 
 13  StreamingTV       7043 non-null   object 
 14  StreamingMovies   7043 non-null   object 
 15  Contract          7043 non-null   object 
 16  PaperlessBilling  7043 non-null   object 


## 1. How many customers are present in the dataset?

In [4]:
data['customerID'].nunique()

7043

Ans1: There are 7043 Unique Customers in the dataset

## 2. What is the distribution of gender among the customers?

In [6]:
data['gender'].value_counts()

Male      3555
Female    3488
Name: gender, dtype: int64

In [7]:
data['gender'].value_counts(normalize=True)

Male      0.504756
Female    0.495244
Name: gender, dtype: float64

Ans2: 50.5% of the customers are Male and 49.5% of the cusomers are female in the given dataset.

## 3. How does age vary among the customers?

In [8]:
data['SeniorCitizen'].value_counts()

0    5901
1    1142
Name: SeniorCitizen, dtype: int64

In [9]:
data['SeniorCitizen'].value_counts(normalize=True)

0    0.837853
1    0.162147
Name: SeniorCitizen, dtype: float64

In [10]:
data['Partner'].value_counts()

No     3641
Yes    3402
Name: Partner, dtype: int64

In [12]:
data['Dependents'].value_counts(normalize=True)

No     0.700412
Yes    0.299588
Name: Dependents, dtype: float64

Ans3 : 16% of customers are seniors citizens i.e; 60+
There is no column on age to view how it varies among customers but 50% customers have partners and 30% have dependents so we can assume they will be of age 21+


## 4. What are the different locations from which customers belong?

Ans: There is no data available on the locations that customer belong to in the given dataset.

## 5. How long have customers been using the service on average?

In [18]:
data['tenure'].describe()

count    7043.000000
mean       32.371149
std        24.559481
min         0.000000
25%         9.000000
50%        29.000000
75%        55.000000
max        72.000000
Name: tenure, dtype: float64

Ans: On Average customers have been using the serivce for 32.4 months. 50% of the customers used the service for atleast 29 months i.e; for two and half years.

## 6. What is the distribution of subscription types?

In [19]:
data['Contract'].value_counts()

Month-to-month    3875
Two year          1695
One year          1473
Name: Contract, dtype: int64

In [21]:
data['Contract'].value_counts(normalize=True)

Month-to-month    0.550192
Two year          0.240664
One year          0.209144
Name: Contract, dtype: float64

Ans: About 55% of the subscription types are month on month. 24% of the subscriptions are Two year subscriptions and 20% of them are one year subscription. More than half are month on month subscirptions but Two year contracts are slightly higher than one year contracts.

## 7. What is the average monthly charge incurred by customers?

In [22]:
data['MonthlyCharges'].mean()

64.76169246059922

Ans: Average monthly charge incurred by the customers are 64.76. But this varies on different services(like:TV, Movies, Internet etc) they have availed and different subsciption types.

## 8. What is the range of total charges accumulated by customers?

In [34]:
#As some of the values in Total Charges column are mentioned with empty string ' ', 
#I've converted them to 0 so we can do numerical analysis on them
data['TotalCharges'] = data['TotalCharges'].apply(lambda x: 0 if x.strip()=='' else float(x))

In [35]:
data['TotalCharges'] = data['TotalCharges'].astype(float)

In [38]:
data['TotalCharges'].describe()

count    7043.000000
mean     2279.734304
std      2266.794470
min         0.000000
25%       398.550000
50%      1394.550000
75%      3786.600000
max      8684.800000
Name: TotalCharges, dtype: float64

Ans: Range of total charges accumulated are from 0 to 8684.8 . 50% of the customers have total charges of at leasst 1394.55 and average value of total charges of all customers is 2279.73

## 9. How many customers have churned?

In [42]:
len(data[data.Churn=='Yes'])

1869

Ans: 1869 ccustomers from 7043 are churned

## 10. What is the churn rate in the dataset?

In [43]:
len(data[data.Churn=='Yes'])*100/len(data)

26.536987079369588

Ans: 26.5% is the churn rate of the given dataset

## 11. How does churn vary across different genders?

In [50]:
pd.crosstab(index=data['gender'],columns=data['Churn'])

Churn     No  Yes
gender           
Female  2549  939
Male    2625  930

In [51]:
pd.crosstab(index=data['gender'],columns=data['Churn'],normalize='index')

Churn         No       Yes
gender                    
Female  0.730791  0.269209
Male    0.738397  0.261603

Ans: We've already seen the distribution between geneders is almost 50% and churn in female can be seen at 26.9% and male at 26.1%. This is also almost equal between both genders and the total churn rate is 26.5%.

## 12. Are there any differences in churn based on the customer's age?

In [53]:
pd.crosstab(index=data['SeniorCitizen'],columns=data['Churn'])

Churn            No   Yes
SeniorCitizen            
0              4508  1393
1               666   476

In [54]:
pd.crosstab(index=data['SeniorCitizen'],columns=data['Churn'],normalize='index')

Churn                No       Yes
SeniorCitizen                    
0              0.763938  0.236062
1              0.583187  0.416813

Ans: Churn in Senior Citizens is 41.7% this is pretty high considering the total churn rate is 26.5%. We can infer that Senior Citizens are most likely to churn than other customers.

## 13. Are customers from specific locations more likely to churn?

Ans: As we dont have much idea about the location of the cutomer this will be difficult to check

## 14. Does the subscription type influence customer churn?

In [56]:
pd.crosstab(index=data['Contract'],columns=data['Churn'])

Churn             No   Yes
Contract                  
Month-to-month  2220  1655
One year        1307   166
Two year        1647    48

In [57]:
pd.crosstab(index=data['Contract'],columns=data['Churn'],normalize='index')

Churn                 No       Yes
Contract                          
Month-to-month  0.572903  0.427097
One year        0.887305  0.112695
Two year        0.971681  0.028319

Ans: Churn rate in Month to Month subscription customers is 42.7%. This is very high considering the churnrate of total dataset is 26.5%. We can infer from this that Month to Month Subscribers are the most likely to churn than one year or two year subscribed customers.

## 15. Is there any correlation between monthly charges and total charges?

In [58]:
data.corr()

SeniorCitizen    tenure  MonthlyCharges  TotalCharges
SeniorCitizen        1.000000  0.016567        0.220173      0.103006
tenure               0.016567  1.000000        0.247900      0.826178
MonthlyCharges       0.220173  0.247900        1.000000      0.651174
TotalCharges         0.103006  0.826178        0.651174      1.000000

Ans: Total Charges have high correlation with Tenure and Monthly charges. This makes sense as Total charges dependeds on number of months the serivce has been availed and charge per month.